airbyte/datasets/_base.py

```diff
@@ -37,6 +37,13 @@ def to_pandas(self) -> DataFrame:
         # duck typing is correct for this use case.
         return DataFrame(cast(Iterator[dict[str, Any]], self))
 
+    def to_arrow_dataset(self, chunksize: int = 100000) -> Any:
```

Comment:
Regarding name here, I think we can abbreviate slightly, to simply `to_arrow()`. I think we can still use the more explicit `get_arrow_dataset()` at the Cache level, which is similar to the pairing of `Cache.get_pandas_dataframe()` and `Dataset.to_pandas()`.